### ISSUE TITLE
Sheep example - POST not working properly?

### ISSUE THREAD

**Issue by JeffHoover** [babelouest/ulfius#7]
In the sheep example, the PUT behaves the way I expect, incrementing `nbsheep` and returning JSON like:

`{
  "nbsheep": 4
}`

The POST, however, doesn't seem to. When I send a POST with a payload like:
`{
  "nbsheep": 6
}`
returns this JSON:
`{
"nbsheep": 0
}`

If, after line 157 in sheep_counter.c, I put `printf("json_body = %s\n", request->json_body);`  and send the POST again, it tells me that the `request->json_body` is null, which would explain why I don't get a value back in the response.
If I assign a constant value into `*nb_sheep` after the `if/else` block, and do another POST, the constant value is returned, as expected, in the JSON response.

**Comment by babelouest** [comment]
The `POST` callback function is designed to initialize the `sheep_counter` to 0 or to the value specified, but the input parameters must be sent in json format, which mean having the header `Content-Type` with the value `application/json` in it.
You may have forgotten the content-type header in your POST request.

**Comment by JeffHoover** [comment]
True, I *did* forget to specify the Content-Type in the header of my POST! Thank you!! I'll try that tonight!

--
Jeff Hoover

**Comment by Syntaxerror1943** [comment]
Hi,
have the same problem with the sheep example. But i have set the 'Content-Type: application/json'. I tried it with the firefox client addon RESTClient and with curl:

`curl -X post -H 'Content-Type: application/json' -i 'http://192.168.2.119:7437/sheep' --data '{"nbsheep":5}'`


i got every time as return {"nbsheep":0}

json_nb_sheep seems to be not NULL with a simple printf test. 

Should i set some more header? I got no errors im i set the error pointer for the second value:

`ulfius_get_json_body_request(request, NULL)`

in the error struct the "text" value ist empty and source points to the string `<buffer>`
